### ISSUE TITLE
Investigate "already creating client for key" when registering command consumer

### ISSUE THREAD

**Issue by calohmn** [eclipse/hono#2660]
There have been some integration test failures with an `error registering consumer with the command router service` failure.
The cause here was a `already creating client for key` error when trying to create the AMQP sender link between the AMQP protocol adapter and the Command Router component.

````
2021-05-14T16:56:28.8645075Z 16:56:28.859 [main] INFO  o.e.h.t.amqp.CommandAndControlAmqpIT - running testSendCommandFailsForCommandRejectedByDevice(AmqpCommandEndpointConfiguration, VertxTestContext) [3]; parameters: subscribe as: GATEWAY_FOR_SINGLE_DEVICE, southbound endpoint: command, northbound endpoint: command
[...]
2021-05-14T16:56:29.0529884Z 16:56:29.052 [vert.x-eventloop-thread-0] INFO  o.e.h.t.amqp.CommandAndControlAmqpIT - AMQPS connection to adapter [localhost:49175] established

2021-05-14T16:56:29.1265809Z AMQP2021-05-14 16:56:29,122 INFO  [org.ecl.hon.ada.cli.com.CommandRouterCommandConsumerFactory] (vert.x-eventloop-thread-0) error registering consumer with the command router service [tenant: 67b741f0-9795-4490-9380-f8d7cb95717f, device: 9e9c2880-d89b-4c22-bfbb-6dd35ecbcb43]: org.eclipse.hono.client.ServerErrorException: already creating client for key
2021-05-14T16:56:29.1270949Z AMQP	at org.eclipse.hono.client.impl.CachingClientFactory.getOrCreateClient(CachingClientFactory.java:215)
2021-05-14T16:56:29.1275020Z AMQP	at org.eclipse.hono.client.impl.CachingClientFactory.lambda$getOrCreateClient$3(CachingClientFactory.java:211)
2021-05-14T16:56:29.1278666Z AMQP	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:942)
[...]
2021-05-14T16:56:29.1567089Z AMQP	at java.base/java.lang.Thread.run(Unknown Source)
2021-05-14T16:56:29.1567701Z AMQP
2021-05-14T16:56:29.1569387Z 16:56:29.145 [vert.x-eventloop-thread-0] WARN  i.v.proton.impl.ProtonSessionImpl - Receiver closed with error
2021-05-14T16:56:29.1571888Z io.vertx.core.impl.NoStackTraceThrowable: Error{condition=amqp:precondition-failed, description='Temporarily unavailable', info=null}
````
Debug log output would have been
````
already trying to create a client for [{}] (max retries reached)
````
(number of retries is 3 with 20ms delay in between)

It should be investigated in how far such error situations can be prevented.

**Comment by sophokles73** [comment]
@calohmn is there anything to be done here?

**Comment by calohmn** [comment]
Not necessarily for 1.10, FMPOV.
I saw this kind of error in a (non-system test) scenario in an mqtt adapter that was being shut down and where lots of "unregister command consumer" requests were about to get sent because of that (see also #2760). 
The command router was restarted at the same time, so there were connection errors/delays happening.

I think the code can be improved here, piggybacking the client creation request on the already ongoing one, instead of doing 3 retries with a delay in between.
Concerning the scenarios in which such errors may happen, I think it's more important to fix #2760 first, preventing one such scenario.
